tuplex/core/src/ee/local/LocalBackend.cc

```diff
@@ -766,6 +766,9 @@ namespace tuplex {
             merge_except_rows = false;
         }
 
+        if (tstage->outputMode() == EndPointMode::FILE) {
```

Comment:
this function probably should be called 2x: Before actually creating the physical plan (so in Logical Plan) and then during execution for the event that the filesystem changed during execution (prob. setting a flag to emit a warning if that was the case is a good idea!).

Reply:
done. Called in toFile and in LocalBackend